### PR TITLE
Feature/spec4.0

### LINKED ISSUE #35: Changing endpoints to new specs

### PR DESCRIPTION
**Pull request by ArturChmura**
Closes #35

### CODE REVIEW

**Comment by ArturChmura** [comment]
Closes #35

File: BikeRentalApi/urls.py [hunk 1]
```diff
@@ -6,19 +6,20 @@
...
     path('stations/blocked/<int:pk>', stations_blocked_detail, name = 'bikes_blocked_detail'),
-    path('register/', auth_views.register, name = 'register'),
-    path('login/', auth_views.login, name = 'login'),
-    path('logout/', auth_views.logout, name = 'logout'),
-    path('techs/', techs_list, name = 'techs_list'),
+    path('register', auth_views.register, name = 'register'),
+    path('login', auth_views.login, name = 'login'),
+    path('logout', auth_views.logout, name = 'logout'),
+    path('techs', techs_list, name = 'techs_list'),
     path('techs/<int:pk>', techs_detail, name = 'techs_detail'),
```
Comment (jankupis) [inline BikeRentalApi/urls.py:24]: This doesn't work with our current frontend, which always appends the trailing slash (and I don't know how to change it)

**Review by jankupis — CHANGES_REQUESTED** [review]

File: BikeRentalApi/urls.py (hunk 1 above)
Comment (ArturChmura) [reply BikeRentalApi/urls.py:24]: But that not backend problem

File: BikeRentalApi/serializers/stationSerializer.py [hunk 2]
```diff
@@ -1,8 +1,20 @@
...
+
     class Meta:
         model = BikeStation
         exclude = ['state']
+
+    def get_status(self, obj: BikeStation):
+        return obj.state.label
+
+    def get_activeBikesCount(self, obj: BikeStation):
+        return Bike.objects.filter(station__pk__exact = obj.pk, bike_state = BikeState.Working).count()
```
Comment (gardzielb) [inline BikeRentalApi/serializers/stationSerializer.py:20]: Can be replaced with `obj.bike_set.all().count()`

File: BikeRentalApi/urls.py (hunk 1 above)
Comment (ArturChmura) [reply BikeRentalApi/urls.py:24]: For it to pass postman tests, trailing slashes must be removed

**Review by gardzielb — CHANGES_REQUESTED** [review]
One small change in parts related with my work, the rest is nice.

File: BikeRentalApi/urls.py (hunk 1 above)
Comment (debskipawel) [reply BikeRentalApi/urls.py:24]: Both backend and frontend need to pass the tests. If the solution to this is not obvious, I think we ought to discuss this more thoroughly in private

Comment (ArturChmura) [reply BikeRentalApi/urls.py]: Never mind, its working in both cases now

File: BikeRentalApi/serializers/stationSerializer.py (hunk 2 above)
Comment (ArturChmura) [reply BikeRentalApi/serializers/stationSerializer.py]: changed to `obj.bike_set.filter(bike_state = BikeState.Working).count()` to return anly avalible bikes

**Review by gardzielb — APPROVED** [review]
Seems nice

**Review by jankupis — APPROVED** [review]
Very good now 🦾

**Pull request merged**